### DISCUSSION TITLE
About Oracle

### DISCUSSION THREAD

**Discussion by yangmeng66** [sqlancer/sqlancer#531] [Q&A]
Excuse me, when will Oracle test be supported?

**Comment by oraluben** [comment]
I suspect Oracle is not likely to be supported, for it's closed-source, and thus has a long time for a reported bug to be fixed, which is a critical issue for fuzzing-based methodology.

**Reply by yangmeng66** [reply]
hello,
Is there more code documentation about "sqlancer"? I want to develop sqlancer of our own database (similar to Oracle)

**Reply by oraluben** [reply]
I don't think so, but you can checkout the initial commit of some newly-added database support (e.g. the nearest one I can find is https://github.com/sqlancer/sqlancer/commit/3f29b049d784cc3b6491aa7f9f767672f876dd83, for oceanbase) to know how to get started, and adapt them to your own database.

**Reply by yangmeng66** [reply]
Hello,
I'm ready to run sqlancer in idea, but I find errors. Can you give me some guidance? Thank you very much

`java: Internal compiler error: java.lang.Exception: java.lang.NoClassDefFoundError: com/google/common/collect/AbstractMapBasedMultimap$KeySet at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.handleProcessor(RoundDispatcher.java:172)
`

**Reply by oraluben** [reply]
Maven project should be supported without additional configuration in IDEA. If you have issue running, you can try command line first, like https://github.com/sqlancer/sqlancer#getting-started. You can also join the Slack channel for use and implement -wise questions: https://github.com/sqlancer/sqlancer#community

**Reply by yangmeng66** [reply]
There is no problem running under Linux

**Reply by oraluben** [reply]
Seems related to IDEA configs:
https://www.cnblogs.com/jpfss/p/9126186.html
https://intellij-support.jetbrains.com/hc/en-us/community/posts/360000177844-Eclipse-compiler-raises-NoClassDefFoundError-for-spring-boot-configuration-processor-s-ConfigurationMetadata-class
